tests/regression/test_tweedie_deviance.py

```diff
@@ -0,0 +1,106 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from collections import namedtuple
+from functools import partial
+
+import pytest
+import torch
+from sklearn.metrics import mean_tweedie_deviance
+from torch import Tensor
+
+from tests.helpers import seed_all
+from tests.helpers.testers import BATCH_SIZE, NUM_BATCHES, MetricTester
+from torchmetrics.functional.regression.tweedie_deviance import tweedie_deviance_score
+from torchmetrics.regression.tweedie_deviance import TweedieDevianceScore
+
+seed_all(42)
+
+Input = namedtuple("Input", ["preds", "targets"])
+
+_single_target_inputs1 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE),
+    targets=torch.rand(NUM_BATCHES, BATCH_SIZE),
+)
+
+_single_target_inputs2 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE),
+    targets=torch.rand(NUM_BATCHES, BATCH_SIZE),
+)
```

Comment:
if the metric supports additional dimensions, we should also test that
```suggestion
)

_multi_target_inputs = Input(
    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 5),
    targets=torch.rand(NUM_BATCHES, BATCH_SIZE, 5),
)
```